arrow-libs/resilience/arrow-resilience/src/commonMain/kotlin/arrow/resilience/Schedule.kt

```diff
@@ -412,6 +417,14 @@ public value class Schedule<in Input, out Output>(public val step: ScheduleStep<
 public suspend fun <A> Schedule<Throwable, *>.retry(action: suspend () -> A): A =
```

Comment:
Are you sure that the compiler is able to infer which version of `retry` you are calling? I'm a bit concerned about problems in resolving which one you want. Maybe we can change the other one to `retryInstanceOf`? This would follow the naming conventions in Kotlin (`filterInstanceOf`...)

Reply:
We should have tests, for this, but let's double check. I like to cover these API concerns in the tests as well, so we can also track it never breaks by accident.